ios/SalesforceReact/SFNetReactBridge.m

```diff
@@ -68,7 +72,7 @@ @implementation SFNetReactBridge
     } else {
         request = [SFRestRequest requestWithMethod:method path:path queryParams:queryParams];
     }
-
+    request.requiresAuthentication = !doesNotRequireAuthentication;
```

Comment:
This confuses me a little bit. Why do we have `requiresAuthentication` on `SFRestRequest` and a similar property on `SFRestAPI` (unauthenticated (vs) authenticated)? Do we really need `requiresAuthentication` on the request object now that we have dedicated authenticated and unauthenticated objects at the client level? In other words, there seems to be clear separation at the `SFRestAPI` layer and this property muddles the waters. What do you think, @trooper2013 and @wmathurin?

Reply:
@bhariharan Perhaps I can add a flag to the unauthenticated global instance of SFRestAPI and force override the value on all requests. But note that the property  ``requiresAuthentication `` is an attribute on the ``SFRestRequest``,   essentially setting this value for a request being invoked on a authenticated instance too. So it is a similar problem? How about removing the property in future versions?

Reply:
I was suggesting that we deprecate the property `requiresAuthentication` from `SFRestRequest` and recommend that developers use the unauthenticated instance from `SFRestAPI` instead. We can then remove that property from the request in `8.0`.